september 2024 news (#849)
ropensci/roweb3: content/blog/2024-09-27-news-september-2024/index.Rmd

  }
  glue::glue_collapse(..., sep = ", ", last = ", and ")
}
```
<!-- Before sending DELETE THE INDEX_CACHE and re-knit! -->

Dear rOpenSci friends, it's time for our monthly news roundup!
<!-- blabla -->
You can read this post [on our blog](`r url`).
Now let's dive into the activity at and around rOpenSci!

## rOpenSci HQ

### R-universe list pages

R-universe now features a [list of all datasets in all packages](https://r-universe.dev/datasets) and a [list of all vignettes  from CRAN, Bioconductor, and others](https://r-universe.dev/articles).

Another important global table is the [package scoreboard](https://r-universe.dev/scores) that lets you explore the scores used by R-universe search engine for ranking, which is based on:

- Stars: Number of starson Github;
- Downloads per month from CRAN or Bioconductor mirrors;
- Scripts: (NEW) number of files on GitHub that mention 'library(pkgname)';
- Dependents: recursive reverse dependencies;
- Unique contributors;
- Yearly commits.

### Resources from our training sessions

Our website now features a searchable table of our [past training sessions](/training/).
Find more of our [resources](/resources).

> 

Reviewer: ```suggestion
## Noam Ross, rOpenSci and communities empowering open science

DataLatam Podcast interviewed Noam. Learn about his work at the intersection of health and the environment and his journey with rOpenSci.  https://www.datalatam.com/109/
```

### The rOpenSci community at upcoming events

> 

Reviewer: ```suggestion
[Will Landau](/author/will-landau/) will deliver a keynote talk at [LatinR <- Latinamerican Conference About the Use of R in R&D](/events/latinr-2024/). The conference is happening online from November 18-22. 

```

### The rOpenSci community at upcoming events

rOpenSci Community Manager [Yanina Bellini Saibene](/author/yanina-bellini-saibene/) will deliver a keynote talk at [BioNT Community Event & CarpentryConnect-Heidelberg 2024](/events/carpentryconnect-2024/), on November 14th in Heidelberg, Germany.

[Will Landau](/author/will-landau/) will deliver a keynote talk at [LatinR (Latinamerican Conference About the Use of R in R&D)](/events/latinr-2024/). The conference is happening online from November 18-22. 

### Coworking

Read [all about coworking](/blog/2023/06/21/coworking/)!

Join us for social coworking & office hours monthly on first Tuesdays! 
Hosted by Steffi LaZerte and various community hosts. 
Everyone welcome. 
No RSVP needed. 
Consult our [Events](/events) page to find your local time and how to join.

- Tuesday, October 1st, 14h00 Europe Central (12:00 UTC), Theme TBA with cohost TBA and [Steffi LaZerte](/author/steffi-lazerte).

And remember, you can always cowork independently on work related to R, work on packages that tend to be neglected, or work on what ever you need to get done!

> 

Reviewer: ```suggestion

## Screen Reader Accessible Tools and Resources for Learning and Working with R
We published the webinar recording that walks through learning and using R with screen readers—and it's bilingual (English & Turkish)! with subtitles in English.

📹 Webinar video featuring [Liz Hare, PhD](/author/liz-hare/)
and [Alican Cagri Gokcek](/author/alican-cagri-gokcek/): https://vimeo.com/1008631708

Also, don't miss the detailed technical note with all the necessary resources to get started. By Liz Hare: https://ropensci.org/blog/2024/09/05/screen-readers-tools/
```

---
slug: "news-september-2024"
title: rOpenSci News Digest, September 2024
author:
  - The rOpenSci Team
date: '2024-09-27'
tags:
  - newsletter

> description: R-universe list pages, training resources, coworking, new package and package news

Reviewer 2: ```suggestion
description: R-Universe list pages, training resources, coworking, new package and package news
```

params:
  last_newsletter: "2024-08-30"
---

```{r setup, include=FALSE}
library("magrittr")
library("rlang")
last_newsletter <- anytime::anytime(params$last_newsletter)
knitr::opts_chunk$set(echo = FALSE)
url <- sprintf(
    "/blog/%s/%s/%s/%s",
    lubridate::year(rmarkdown::yaml_front_matter(knitr::current_input())$date),
    stringr::str_pad(lubridate::month(rmarkdown::yaml_front_matter(knitr::current_input())$date), 2, "0", side = "left"),
    stringr::str_pad(lubridate::day(rmarkdown::yaml_front_matter(knitr::current_input())$date), 2, "0", side = "left"),
    rmarkdown::yaml_front_matter(knitr::current_input())$slug
    )
english <- function(x) {
  as.character(english::english(x))
}

nice_string <- function(...) {
  if (length(...) == 2) {
    return(paste(c(...), collapse = " and "))
  }
  glue::glue_collapse(..., sep = ", ", last = ", and ")
}
```
<!-- Before sending DELETE THE INDEX_CACHE and re-knit! -->

Dear rOpenSci friends, it's time for our monthly news roundup!
<!-- blabla -->
You can read this post [on our blog](`r url`).
Now let's dive into the activity at and around rOpenSci!

## rOpenSci HQ

> ### R-universe list pages

Reviewer 2: ```suggestion
### R-Universe list pages
```

> R-universe now features a [list of all datasets in all packages](https://r-universe.dev/datasets) and a [list of all vignettes  from CRAN, Bioconductor, and others](https://r-universe.dev/articles).

Reviewer 2: ```suggestion
R-Universe now features a [list of all datasets in all packages](https://r-universe.dev/datasets) and a [list of all vignettes  from CRAN, Bioconductor, and others](https://r-universe.dev/articles).
```

> Another important global table is the [package scoreboard](https://r-universe.dev/scores) that lets you explore the scores used by R-universe search engine for ranking, which is based on:

Reviewer 2: ```suggestion
Another important global table is the [package scoreboard](https://r-universe.dev/scores) that lets you explore the scores used by R-Universe search engine for ranking, which is based on:
```

> - Stars: Number of starson Github;

Reviewer 2: ```suggestion
- Stars: Number of stars on GitHub;
```

- Downloads per month from CRAN or Bioconductor mirrors;
- Scripts: (NEW) number of files on GitHub that mention 'library(pkgname)';
- Dependents: recursive reverse dependencies;
- Unique contributors;
- Yearly commits.

> ### Resources from our training sessions

Reviewer 2: At first I thought this still referred to the R-Universe stuff from above, so perhaps add rOpenSci to clarify?
By the way, this is so cool!
```suggestion
### Resources from rOpenSci training sessions
```

> Our website now features a searchable table of our [past training sessions](/training/).
> Find more of our [resources](/resources).

Reviewer 2: ```suggestion
We've added a searchable table of [past training sessions](/training/) to our resources hosted on the rOpenSci website.
Explore the full list of our [resources](/resources).
```

> ### Noam Ross, rOpenSci and communities empowering open science

Reviewer 2: ```suggestion
### Noam Ross, rOpenSci, and communities empowering open science
```

DataLatam Podcast [interviewed](https://www.datalatam.com/109/) rOpenSci director Noam Ross. 
Learn about his work at the intersection of health and the environment and his journey with rOpenSci.

> ### Resources: Screen Reader Accessible Tools and Resources for Learning and Working with R

Reviewer 2: Otherwise it sounds like it should be listed above with the training materials. 
```suggestion
### Materials: Screen Reader Accessible Tools and Resources for Learning and Working with R
```

We have now published the recording of the webinar that walks through learning and using R with screen readers — and it's bilingual (English & Turkish)! with subtitles in English.

📹 [Webinar video](https://vimeo.com/1008631708) featuring [Liz Hare, PhD](/author/liz-hare/)
and [Alican Cagri Gokcek](/author/alican-cagri-gokcek/)

> Also, don't miss the [detailed technical note](https://ropensci.org/blog/2024/09/05/screen-readers-tools/) by Liz Hare with all the necessary resources to get started.

Reviewer 2: ```suggestion
Also, don't miss the [detailed technical note](/blog/2024/09/05/screen-readers-tools/) by Liz Hare with all the necessary resources to get started.
```

### The rOpenSci community at upcoming events

rOpenSci Community Manager [Yanina Bellini Saibene](/author/yanina-bellini-saibene/) will deliver a keynote talk at [BioNT Community Event & CarpentryConnect-Heidelberg 2024](/events/carpentryconnect-2024/), on November 14th in Heidelberg, Germany.

> [Will Landau](/author/will-landau/) will deliver a keynote talk at [LatinR (Latinamerican Conference About the Use of R in R&D)](/events/latinr-2024/). The conference is happening online from November 18-22. 

Reviewer 2: ```suggestion
[Will Landau](/author/will-landau/) will deliver a keynote talk at [LatinR (Latin American Conference About the Use of R in R&D)](/events/latinr-2024/). The conference is happening online from November 18-22. 
```

### Coworking

Read [all about coworking](/blog/2023/06/21/coworking/)!

Join us for social coworking & office hours monthly on first Tuesdays! 
Hosted by Steffi LaZerte and various community hosts. 
Everyone welcome. 
No RSVP needed. 
Consult our [Events](/events) page to find your local time and how to join.

> - Tuesday, October 1st, 14h00 Europe Central (12:00 UTC), Theme TBA with cohost TBA and [Steffi LaZerte](/author/steffi-lazerte).
> 
> And remember, you can always cowork independently on work related to R, work on packages that tend to be neglected, or work on what ever you need to get done!

Reviewer 2: ```suggestion
- [Tuesday, October 1st, 14h00 Europe Central (12:00 UTC)](/events/coworking-2024-10), Theme Using R to tell (data science) stories with cohost [Apoorv Anand](/author/apoorv-anand) and [Steffi LaZerte](/author/steffi-lazerte).
    - Explore how others use R to tell data stories
    - Start work on a project using R to craft a narrative through data
    - Chat with our cohost to discuss and explore the use of R in data stories
- [Tuesday, November 5th, 09:00 Americas Pacific (17:00 UTC)](/events/coworking-2024-11), Theme Getting to know Openscapes with cohost [Stefanie Butland](/author/stefanie-butland) and [Steffi LaZerte](/author/steffi-lazerte).
    - Look up [Openscapes](https://openscapes.org/)
    - Chat with our cohost to learn more about the Openscapes organization, and how they use R to automate creating open education infrastructure

And remember, you can always cowork independently on work related to R, work on packages that tend to be neglected, or work on what ever you need to get done!
```

* [historydata](https://docs.ropensci.org/historydata/), datasets for historians. [Issue for volunteering](https://github.com/ropensci/historydata/issues/23).

### Calls for contributions

* [Help make qualtRics better! Code contributions wanted](/blog/2024/03/01/qualtrics-call-help/)

* [Help make assertr better! Come close issues](/blog/2024/02/27/assertr-call-help/)

* [Help users and developers of targets by answering questions!](/blog/2024/02/29/targets-call-help/)

* [Help make waywiser better! User requests wanted](/blog/2024/02/19/waywiser-call-help/)

Also refer to our [help wanted page](/help-wanted/) -- before opening a PR, we recommend asking in the issue whether help is still needed.

## Package development corner

Some useful tips for R package developers. :eyes:

### Reminder: R Consortium Infrastructure Steering Committee (ISC) Grant Program Accepting Proposals until October 1st!

The R Consortium [Call for Proposals](https://www.r-consortium.org/all-projects/call-for-proposals) might be a relevant funding opportunity for your package!

> If you can't prepare your proposal in time, the next call should start [March 1st next year](https://www.r-consortium.org/all-projects/call-for-proposals).

Reviewer 2: ```suggestion
If you can't prepare your proposal in time, the next call should start [March 1st 2025](https://www.r-consortium.org/all-projects/call-for-proposals).
```

### Create and use a custom roxygen2 tag

Read more about creating a custom tag that adds a new section to a manual page in this [blog post by Maëlle Salmon](https://masalmon.eu/2024/09/03/roxygen2-custom-tag/).

### Materials from posit::conf(2024) workshops

Materials from posit::conf(2024) workshops have been [posted](https://posit-conf-2024.github.io/workshops/).
Especially relevant to package developers: [From R User to R Programmer](https://posit-conf-2024.github.io/programming-r/) by Emma Rand and Ian Lyttle; and [Package Development: The Rest of the Owl](https://github.com/posit-conf-2024/pkg-dev) by Jenny Bryan whose introduction slidedeck features GitHub search tips.

### Some tips when searching GitHub issues

Speaking of GitHub search...

Michael Chirico shared the following tip on Mastodon: ["[you can use] the 'in:title' operator in GitHub search to restrict a search to issues/PRs matching a query in the title"](https://fosstodon.org/@michaelchirico/113083595380080927).

Jenny Bryan answered with another tip: ["involves:USERNAME" for "I was once part of a conversation about this"](https://mastodon.social/@jennybryan@fosstodon.org/113085363392572924).

> ### An usethis function to create a test helper file

Reviewer 2: I felt instinctively that it should be "A" not "An", but had to look it up to be sure. It has to do with how the word sounds, not only whether it's a vowel or not: https://owl.purdue.edu/owl/general_writing/grammar/articles_a_versus_an.html (see the exceptions section).

```suggestion
### A usethis function to create a test helper file
```

The new usethis function [` usethis::use_test_helper("mocks")`](https://usethis.r-lib.org/reference/use_test_helper.html) will create a [helper file](https://testthat.r-lib.org/articles/special-files.html) under `tests/testthat/helper-mocks.R`.

### A devtools function for building the README and vignettes

> Not a new function, but in case you ignore it...

Reviewer 2: ```suggestion
Not a new function, but in case you missed it...
```
